lib/log.js

```diff
@@ -1,31 +1,101 @@
 "use strict";
 
-var chalk = require("chalk");
+// Imports note: this file is the _only_ one required by `bin/builder.js` which
+// can be a global or local import. To avoid using old versions of libs before
+// we switch to this local scope, we lazy-require things.
 
 // Wrap "type".
 var wrapType = function (type) {
   return "[builder" + (type ? ":" + type : "") + "]";
 };
 
+// Levels
+var LEVELS = {
+  info: 0,
+  warn: 1,
+  error: 2,
+  none: 3
+};
+
 /**
  * A super-small logger.
  */
-module.exports = {
-  // TODO(6): Configurable log levels.
-  // https://github.com/FormidableLabs/builder/issues/6
-  _logger: function () {
-    return console;
+var log = module.exports = {
+  /**
+   * Set log level from command flags or environment.
+   *
+   * **NOTE - Side Effects**: Mutates environment.
+   *
+   * @param {Object} opts         Options object
+   * @param {Object} opts.argv    Arguments array.
+   * @param {Object} opts.env     Environment object to mutate (Default `process.env`)
+   * @returns {void}
+   */
+  setLevel: function (opts) {
+    opts = opts || {};
+    var env = opts.env && opts.env.env || process.env;
+
+    // Try to determine log level from environment.
+    var level = env._BUILDER_ARGS_LOG_LEVEL;
+
+    // If not, lazy require args and determine log level from command line.
+    if (!level) {
+      var args = require("./args"); // eslint-disable-line global-require
+      var parsed = args.general(opts.argv);
+      level = parsed.quiet === true ? "none" : parsed.logLevel;
+    }
+
+    // Statefully set level.
+    env._BUILDER_ARGS_LOG_LEVEL = level;
+    log._level = LEVELS[level];
+    if (typeof log._level === "undefined") {
+      throw new Error("Unknown log level: " + level);
+    }
+
+    // Drain message queue.
+    if (log._queue) {
+      log._queue.forEach(function (obj) {
+        log[obj.level](obj.type, obj.msg);
+      });
+      log._queue = null;
+    }
   },
 
-  info: function (type, msg) {
-    this._logger().info([chalk.green(wrapType(type)), msg].join(" "));
+  // Nuke everything for test runs.
+  _unsetLevel: function () {
+    delete process.env._BUILDER_ARGS_LOG_LEVEL;
+    delete log._level;
+    delete log._queue;
   },
 
-  warn: function (type, msg) {
-    this._logger().warn([chalk.yellow(wrapType(type)), msg].join(" "));
+  _logger: function () {
+    return console;
   },
 
-  error: function (type, msg) {
-    this._logger().error([chalk.red(wrapType(type)), msg].join(" "));
+  _wrapper: function (level, color, type, msg) { // eslint-disable-line max-params
+    // Queue if level is unset.
+    if (typeof log._level === "undefined") {
+      log._queue = log._queue || [];
```

Comment:
Above you're setting `log._queue = null`. Instead why don't you just drain the queue by doing:
```
log._queue = map(function (obj) {
  log[obj.level](obj.type, obj.msg)
});
```

which will set it to `[]` since that's what you want anyway.